src/bandersnatch_storage_plugins/s3.py

```diff
@@ -0,0 +1,368 @@
+import configparser
+import contextlib
+import datetime
+import hashlib
+import logging
+import os
+import pathlib
+from pathlib import PurePath
+import tempfile
+from typing import (
+    IO,
+    Any,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Type,
+    Union,
+    TextIO
+)
+
+import boto3
+from botocore.client import Config
+import filelock
```

Comment:
Does this get used?

Reply:
yes, S3FileLock is based on `filelock.BaseFileLock`